src/test/java/com/xceptance/neodymium/util/SelenideAddonsTest.java

```diff
@@ -237,125 +303,130 @@ public void testWrapAssertionErrorWithoutMessage()
-                     });
-        runArray.add(
-                     () -> {
-                         throw new StaleElementReferenceException("You shall not pass!");
-                     });
-        runArray.add(
-                     () -> {
-                         throw new StaleElementReferenceException("You shall not pass!");
-                     });
-        runArray.add(
-                     () -> {
-                         throw new StaleElementReferenceException("You shall pass!");
-                     });
-        runArray.add(
-                     () -> {
-                         throw new StaleElementReferenceException("You shall never be seen!");
-                     });
         final Iterator<Runnable> iterator = runArray.iterator();
 
         // testing the error path after three exceptions
         openBlogPage();
         long startTime = new Date().getTime();
         try
         {
-            SelenideAddons.$safe(() -> {
-                counter.incrementAndGet();
-                if (iterator.hasNext())
-                {
-                    iterator.next().run();
-                }
-                return $("body").should(exist);
-            });
+            if (isSupplier)
+            {
+                testSupplier(counter, iterator);
+            }
+            else
+            {
+                testRunnable(counter, iterator);
+            }
         }
-        catch (StaleElementReferenceException e)
+        catch (StaleElementReferenceException | UIAssertionError e)
         {
-            Assert.assertTrue(e.getMessage().startsWith("You shall pass!"));
+            Assert.assertTrue(e.getMessage() + " doesn't contain phrase 'You shall pass!'", e.getMessage().contains("You shall pass!"));
         }
-        long endTime = new Date().getTime();
-        Assert.assertTrue(endTime - startTime > Neodymium.configuration().staleElementRetryTimeout());
-        Assert.assertEquals(counter.get(), Neodymium.configuration().staleElementRetryCount() + 1);
 
-        // testing the happy path after one exception
-        SelenideElement element = SelenideAddons.$safe(() -> {
-            counter.incrementAndGet();
-            if (iterator.hasNext())
-            {
-                iterator.next().run();
-            }
-            return $("body");
-        });
-        Assert.assertEquals(counter.get(), Neodymium.configuration().staleElementRetryCount() + 3);
-        element.shouldBe(visible);
+        long duration = new Date().getTime() - startTime;
+
+        long minimalDuration = Neodymium.configuration().staleElementRetryTimeout() * Neodymium.configuration().staleElementRetryCount();
+        // real duration for cause check is mostly only few seconds more than minimal duration, although the duration of
+        // message check is always approximately 1000ms more due to UIAssertion wrapping
+        long maximalDuration = minimalDuration + 1000;
```

Comment:
replace the 1000 by staleElementRetryTimeout/2

Reply:
Done